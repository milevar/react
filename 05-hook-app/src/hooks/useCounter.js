import React, {useState} from 'react';


export const useCounter = (initialState = 10) => {
   const [counter, setCounter] = useState(initialState);

   // const increment = (factor = 1) => {
   //     setState(state + factor)
   // }
   //
   //  const decrement = (factor = 1) => {
   //      setState(state - factor)
   //  }

    const reset = () => {
        setCounter(initialState)
    }

    const decrement = () => {
        setCounter(counter - 1)
    }

    const increment = () => {
        setCounter(counter + 1)
    }

    return {
       counter, increment, decrement, reset
    };


};

export default useCounter;
