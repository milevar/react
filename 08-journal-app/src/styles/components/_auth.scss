
.auth_main {
  display: flex;
  background-color: $primary;
  align-items: center;
  justify-content: center;
  margin: 0px;
  height: 100vh;
  width: 100vw;
}

.auth_box-container {
  background-color: white;
  box-shadow: 0px 3px $dark-grey;
  border-radius: 2px;
  padding: 20px;
  width: 300px;
}

.auth_title {
  color: darken($color: $primary, $amount: 20);
  margin-bottom: 20px;
}

.auth_input {
  color: $dark-grey;
  border: 0px;
  border-bottom: 1px solid solid $light-grey;
  margin-bottom: 10px;
  height: 20px;
  width: 100%;

  transition: border-bottom .3s ease;
  &:focus {
    border-bottom: 1px solid $primary;
    outline: none;
  }
}

.auth_social-networks {
  align-items: center;
  display: flex;
  justify-content: center;
  flex-direction: column;
  padding-top: 20px;
  padding-bottom: 20px;
  width: 100%;
}

.auth_alert-error {
  background-color: red;
  border-radius: 5px;
  color: white;
  display: flex;
  justify-content: center;
  margin-bottom: 10px;
  padding: 5px;
}
