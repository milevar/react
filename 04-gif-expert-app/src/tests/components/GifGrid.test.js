import React from "react";
import {shallow} from "enzyme";
import {GifGrid} from "../../components/GifGrid";
import {useFetchGifs} from '../../hooks/useFetchGifs'

jest.mock('../../hooks/useFetchGifs')


describe('Pruebas en <GifGrid />', () => {

    const category = 'Garfield';

    test('Debe de mostrar el componente correctamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category}/>)
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de mostrar items cuando se cargan imagenes', () => {

        const gifs = [
            {
                id: 'ABC',
                url: 'http://localhost/cualquier/cosa.jpg',
                title: 'Test 123'
            },
            {
                id: 'ABC',
                url: 'http://localhost/cualquier/cosa.jpg',
                title: 'Test 123'
            }
        ];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category}/>)

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    })


})


