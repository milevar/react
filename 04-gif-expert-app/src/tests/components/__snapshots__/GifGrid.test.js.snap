// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Garfield
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid /> Debe de mostrar items cuando se cargan imagenes 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Garfield
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Test 123"
      url="http://localhost/cualquier/cosa.jpg"
    />
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Test 123"
      url="http://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;
